### DISCUSSION TITLE
Add a torch nn custom model for forecasting time series

### DISCUSSION THREAD

**Discussion by jmren168** [autogluon/autogluon#3407] [Q&A] [answered]
Hi @shchur,

Following the discussion "[Add a custom model for forecasting time series](https://github.com/autogluon/autogluon/discussions/3091)", it looks like I could add a traditional machine learning model for forecasting time series in an ensemble way. 

If so, could I add a torch nn model (e.g., LSTM) for forecasting time series without in an ensemble way?  If I follow the structure of AbstractTimeSeriesModel, what should I do next? Any comments are highly appreciated. 

Best,

**Comment by shchur** [comment]
Hi, after following all instructions from #3091, you should be able to train the model as
```python
class MyCustomModel(AbstractTimeSeriesModel):
    def _fit(...):
        ...
    def predict(...):
        ...

predictor = TimeSeriesPredictor(...).fit(
    train_data, 
    hyperparameters={
        MyCustomModel: {},         
        # add other models to include in the ensemble here
        "AutoETS": {},
        "DeepAR": {},
        ...
    }
)
```

**Reply by shchur** [reply, marked as answer]
Please have a look at the API of `AbstractTimeSeriesModel` - your signatures of the `_fit` and `predict` methods should match those of the abstract method.
